docs/antrea-ipam.md

```diff
@@ -46,3 +46,148 @@ cluster. Valid range is 16 to 30. Default is 24.
+according to `IPPool` spec.
+
+Note that the IP pool annotation cannot be updated or deleted without recreating the
+resource. An `IPPool` can be extended, but cannot be shrunk if already assigned to a
+resource. The IP ranges of IPPools must not overlap, otherwise it would lead to undefined
+behavior.
+
+Regular `Subnet per Node` IPAM will continue to be used for resources without the IP pool
+annotation, or when `AntreaIPAM` feature is disabled.
+
+### Usage
+
+#### Enable AntreaIPAM feature gate
+
+Users need to enable `AntreaIPAM` from the featureGates map defined in antrea.yml for
+both Controller and Agent:
+
+```yaml
+  antrea-controller.conf: |
+    ...
+    featureGates:
+    # Enable flexible IPAM mode for Antrea. This mode allows to assign IP Ranges to Namespaces,
+    # Deployments and StatefulSets via IP Pool annotation.
+      AntreaIPAM: true
+    ...
+  antrea-agent.conf: |
+    ...
+    featureGates:
+    # Enable flexible IPAM mode for Antrea. This mode allows to assign IP Ranges to Namespaces,
+    # Deployments and StatefulSets via IP Pool annotation.
+      AntreaIPAM: true
+    ...
+```
+
+#### Create IPPool CR
+
+The following example YAML manifests create an IPPool CR.
+
+```yaml
+apiVersion: "crd.antrea.io/v1alpha2"
+kind: IPPool
+metadata:
+  name: pool1
+spec:
+  ipVersion: 4
+  ipRanges:
+  - start: "10.2.0.12"
+    end: "10.2.0.20"
+    gateway: "10.2.0.1"
+    prefixLength: 24
+```
+
+#### IPPool Annotations on Namespace
+
+The following example YAML manifests create a Namespace to allocate Pod IPs from the IP pool.
+
+```yaml
+kind: Namespace
+metadata:
+  annotations:
+    ipam.antrea.io/ippools: 'pool1'
+...
+```
+
+#### IPPool Annotations on Pod (available since Antrea 1.5)
+
+In Antrea 1.5, Pod IPPool annotation is supported and have a higher priority than the
+Namespace IPPool annotation. This annotation can be added to `PodTemplate` of a
+controller resource such as StatefulSet and Deployment.
+
+Pod IP annotation is supported for a single Pod to specify a fixed IP for the Pod.
+
+Examples of annotations on a Pod or PodTemplate:
+
+```yaml
+kind: StatefulSet
+spec:
+  template:
+    metadata:
+      annotations:
+        ipam.antrea.io/ippools: 'sts-ip-pool1'  # This annotation will be set automatically on all Pods managed by this resource
+        # Do not add pod-ips annotation to PodTemplate
+...
+```
+
+```yaml
+kind: Pod
+metadata:
+  annotations:
+    ipam.antrea.io/ippools: 'pod-ip-pool1'
+...
+```
+
+```yaml
+kind: Pod
+metadata:
+  annotations:
+    ipam.antrea.io/ippools: 'pod-ip-pool1'
+    ipam.antrea.io/pod-ips: '<ip-in-pod-ip-pool1>'
+...
+```
+
+```yaml
+kind: Pod
+metadata:
+  annotations:
+    ipam.antrea.io/pod-ips: '<ip-in-namespace-pool>'
+...
+```
+
+#### Persistent IP for StatefulSet Pod (available since Antrea 1.5)
+
+A StatefulSet Pod's IP will be kept when Pod restarts, when the IP is allocated from the
```

Comment:
Maybe change the first "when" to "after" to avoid two "when"s.

Reply:
Fixed.